### PR TITLE
Add support for usage record summary

### PR DESCRIPTION
**Pull request by remi-stripe**
cc @stripe/api-libraries

### CODE REVIEW

**Comment by remi-stripe** [comment]
Okay confirmed the resources can't be expanded (invoice and subscription item) so ready to review.

r? @brandur-stripe

File: usage_record_summary.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package stripe
+
+// UsageRecordSummary represents a usage record summary.
+// See https://stripe.com/docs/api#usage_records
+type UsageRecordSummary struct {
+	ID               string  `json:"id"`
+	Live             bool    `json:"livemode"`
```
Comment (brandur-stripe) [inline usage_record_summary.go]: I think we agreed on the fully expanded `Livemode` for these types of properties, but doh! I just noticed when grepping that we missed a couple places — usage records, issuer fraud records, and source transactions.

I'm not sure what to do with those ones right now, but can we make this one `Livemode`?

File: usage_record_summary.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	Invoice          string  `json:"invoice"`
```
Comment (brandur-stripe) [inline usage_record_summary.go]: Also, I think for most structs we went with pure alphabetical order — `ID` and `Livemode` don't get special treatment.

Again, it looks like usage records are a little weird in this respect.

File: usagerecordsummary/client.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+package usagerecordsummary
+
+import (
+	"net/http"
+
+	stripe "github.com/stripe/stripe-go"
+	"github.com/stripe/stripe-go/form"
+)
+
+// Client is used to invoke /plans APIs.
```
Comment (brandur-stripe) [inline usagerecordsummary/client.go]: I think for most of these we went with something like:

``` go
// Client is used to invoke APIs related to usage record summaries.
```

File: usage_record_summary.go (hunk 1 above)
Comment (remi-stripe) [reply usage_record_summary.go]: Damn nice catch, I copy/pasted from usagerecord and when I double checked I saw other matches and did not look further

**Comment by remi-stripe** [comment]
@brandur-stripe  Nice catch on all this. I took this as an opportunity to fix all `Live` and fix the naming of the files to remove the `_` which kind of indicates a namespace in theory (in our API).

PTAL

**Comment by brandur-stripe** [comment]
LGTM. Thanks!

**Comment by brandur-stripe** [comment]
Released as 42.0.0.

**Pull request merged**